Add capability for roles api in files_sharing

sharidas opened the pull request:
Add capability for roles api in files_sharing

Signed-off-by: Sujith H <sharidasan@owncloud.com>

<!--
Thanks for submitting a change to ownCloud!

This is the bug tracker for the Server component. Find other components at https://github.com/owncloud/core/blob/master/.github/CONTRIBUTING.md#guidelines

For fixing potential security issues please see https://owncloud.org/security/

To make it possible for us to get your change reviewed and merged please carefully fill out the requested information below.

Please note that any kind of change needs first be submitted to the master branch which holds the next major version of ownCloud.

We will carefully discuss if your change can or has to be backported to stable branches.

Please set the following labels:

- Set label "3 - To review" for review or "2 - Development" in case the PR still has open tasks
- Set label "backport-request" if backport is needed
- Assignment: assign to self
- Milestone: set the same as the ticket this PR fixes, or "development" by default
- Reviewers: pick at least one
-->

## Description
<!--- Describe your changes in detail -->
Add capabilty for roles api added in the server. `roles_api` is added to the public array to true, if `shareapi_enabled` is `yes` and `shareapi_allow_links` is `yes` set in the appconfig table.
Updated the change to arrange alignment of roles delivered from the RolesController and adjusted the description of View/Download/Edit in the roles. Updated the alignmnent and description change in the fontend (core/js/sharedialoglinkshareview.js).

## Related Issue
<!--- This project only accepts pull requests related to open issues -->
<!--- If suggesting a new feature or change, please discuss it in an issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here: -->
- Fixes <issue_link>

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
Roles api can now be obtained from the capabilities. The key `roles_api` is added to true, if:
- `shareapi_enabled` is set to `yes`, in appconfig
- `shareapi_allow_links` is set to `yes`, in appconfig.


## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->
- php unit test

## Screenshots (if appropriate):

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:
<!-- Tick the checkboxes when done. -->
<!-- Raise documentation ticket in https://github.com/owncloud/documentation -->
- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:
<!-- In case of incomplete PR, please list the open tasks here -->
- [ ] Backport (if applicable set "backport-request" label and remove when the backport was done)

sharidas committed:
Add capability for roles api in files_sharing

Add capability for roles api in files_sharing

Signed-off-by: Sujith H <sharidasan@owncloud.com>

PVince81 commented on apps/files_sharing/lib/Capabilities.php:
```diff
@@ -85,6 +85,8 @@ public function getCapabilities() {
 				$public['password']['enforced_for']['upload_only'] = $woPasswordEnforced;
 				$public['password']['enforced'] = $roPasswordEnforced || $rwPasswordEnforced || $woPasswordEnforced;
 
+				$public['roles_api_with_share_attributes'] = true;
```
please rename to "roles_api"

sharidas replied:
Done.

PVince81 commented:
@sharidas as discussed we should also swap the permissions in this PR ?

sharidas committed:
Alignment correction in the roles delivered

Alignment correction is required for the roles
delivered. For instance Download/View/Upload
should be ahead of Download/View/Edit.
The description for Download/View/Edit is also
corrected in this change.

Signed-off-by: Sujith H <sharidasan@owncloud.com>

sharidas commented:
> @sharidas as discussed we should also swap the permissions in this PR ?

I have added a second commit to this PR. I was preparing the same and missed to add to this PR. Now I have added as  second commit.

PVince81 commented:
also needs swapping in the web UI accordingly ?

sharidas committed:
Adjust the alignment in the webUI for public links

Adjust the alignment of view and upload for
public link dialog window.

Signed-off-by: Sujith H <sharidasan@owncloud.com>

sharidas commented:
> also needs swapping in the web UI accordingly ?

The swapping in the web UI is done.

phil-davis commented on apps/files_sharing/tests/CapabilitiesTest.php:
```diff
@@ -120,6 +120,29 @@ public function testNoLinkSharing() {
 		$this->assertFalse($result['public']['enabled']);
 	}
 
+	public function providesRolesCapbility() {
```
```suggestion
	public function providesRolesCapability() {
```

sharidas replied:
Done

phil-davis commented on apps/files_sharing/tests/CapabilitiesTest.php:
```diff
@@ -120,6 +120,29 @@ public function testNoLinkSharing() {
+	}
+
+	/**
+	 * @dataProvider providesRolesCapbility
```
```suggestion
	 * @dataProvider providesRolesCapability
```

sharidas replied:
Done

PVince81 reviewed (approved):
:+1:
